File: RobotReturnToOrigin.py
# Robot Return to Origin
# - Given a string `moves` representing a sequence of moves by a robot on a 2D plane.  
# - Valid moves: 'R' (right), 'L' (left), 'U' (up), 'D' (down).  
# - Return `True` if the robot returns to the origin (0,0) after all moves, otherwise `False`.  

class Solution:
    def judgeCircle(self, moves: str) -> bool:
        x = 0
        y = 0
        for move in moves:
            if move == 'U':
                y += 1
            elif move == 'D':
                y -= 1
            elif move == 'L':
                x -= 1
            elif move == 'R':
                x += 1
        return x == 0 and y == 0

File: RobotReturnToOrigin.cs
// Robot Return to Origin
// - Check if a sequence of moves returns the robot to (0,0).
// - Valid moves: 'R', 'L', 'U', 'D'

using System;

class Solution
{
    public bool JudgeCircle(string moves)
    {
        int x = 0, y = 0; 

        foreach (char move in moves)
        {
            if (move == 'U') y += 1;
            else if (move == 'D') y -= 1;
            else if (move == 'L') x -= 1;
            else if (move == 'R') x += 1;
        }

        return x == 0 && y == 0; 
    }
}
